### DISCUSSION TITLE
How configurate the client name of wifi station on esp32?

### DISCUSSION THREAD

**Discussion by jeffer80** [nanoframework/Home#1274] [Q&A] [answered]
I connect to wifi router with the below codes:
```
WifiAdapter wifiAdapter = WifiAdapter.FindAllAdapters()[0];
adapter.Connect(SSID, WifiReconnectionKind.Automatic, WIFI_PASS);
```

The client name shown on the router's dhcp leases list is "nanodevice_xxxxxx",
How can I customize the client name?

**Comment by alberk8** [comment]
The only way is to build your own firmware and with changes to the  [config file](https://github.com/nanoframework/nf-interpreter/blob/main/targets/ESP32/_IDF/sdkconfig.default_nopsram.esp32) but that still append the _XXXXXX which the last six characters of the mac address.

CONFIG_LWIP_LOCAL_HOSTNAME="nanodevice"

**Reply by jeffer80** [reply]
Thanks a lot, I just want to change the 'nanodevice' to the name of my product only.  
Is it possible that we can rewrite the value of the whole name in c# in the future version?

**Reply by josesimoes** [reply, marked as answer]
Not likely... Because that's in the network layer which is running before the CLR starts executing C# application. That would require a network "restart"...

**Reply by Ellerbach** [reply]
To be more precise, it's possible but yes, requires a restart of the network stack, see here: https://github.com/espressif/esp-idf/blob/df9310ada26123d8d478bcfa203f874d8c21d654/components/esp_netif/include/esp_netif.h#L397

And note that the network is started already at boot time, it's the reason why you get connected to wifi almost instantaneously once the device gets connected. So changing at run time the hostname will require a restart of the stack, you'll "loose" some time.

If you go for your own products, you may want to flash a specific version with specific names and adjust them. That would be the way I'll go.
